app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/actions/SessionDataActionWithRetrievedData.scala

```diff
@@ -41,7 +41,9 @@ class SessionDataActionWithRetrievedData @Inject()(val sessionStore: SessionStor
   def sessionDataAction[A](
     sessionData: Option[SessionData],
     request: AuthenticatedRequestWithRetrievedData[A]
-  ): RequestWithSessionDataAndRetrievedData[A] =
-    RequestWithSessionDataAndRetrievedData(sessionData, request)
+  ): RequestWithSessionDataAndRetrievedData[A] = {
+    val data: SessionData = sessionData.getOrElse(SessionData.empty).copy(userType = request.userType)
+    RequestWithSessionDataAndRetrievedData(Some(data), request)
```

Comment:
it doesn't look like `RequestWithSessionDataAndRetrievedData` needs to have `SessionData` optional in this case

Reply:
No it doesn't - that was your original design. Do you want to me to change it?

Reply:
do we still need the line above it to populate the user type if the StartController is explicitly populating the user type now?